src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/scalable/LabeledVariantAnnotationsWalker.java

```diff
@@ -303,57 +326,72 @@ final List<Triple<List<Allele>, VariantType, TreeSet<String>>> extractVariantMet
-                                                          final FeatureContext featureContext) {
-        final TreeSet<String> overlappingResourceLabels = new TreeSet<>();
+    private TreeSet<String> findMatchingResourceLabels(final VariantContext vc,
+                                                       final Allele altAllele,
+                                                       final FeatureContext featureContext) {
+        final TreeSet<String> matchingResourceLabels = new TreeSet<>();
         for (final FeatureInput<VariantContext> resource : resources) {
             final List<VariantContext> resourceVCs = featureContext.getValues(resource, featureContext.getInterval().getStart());
             for (final VariantContext resourceVC : resourceVCs) {
-                if (useASAnnotations && !doAllelesMatch(refAllele, altAllele, resourceVC)) {
+                if (useASAnnotations && !doAllelesMatch(vc.getReference(), altAllele, resourceVC)) {
                     continue;
                 }
-                if (isValidVariant(vc, resourceVC, !doNotTrustAllPolymorphic)) {
+                if (isMatchingVariant(vc, resourceVC, !doNotTrustAllPolymorphic, resourceMatchingStrategy)) {
                     resource.getTagAttributes().entrySet().stream()
                             .filter(e -> e.getValue().equals("true"))
                             .map(Map.Entry::getKey)
-                            .forEach(overlappingResourceLabels::add);
+                            .forEach(matchingResourceLabels::add);
                 }
             }
         }
-        return overlappingResourceLabels;
+        return matchingResourceLabels;
     }
 
-    private static boolean isValidVariant(final VariantContext vc,
-                                          final VariantContext resourceVC,
-                                          final boolean trustAllPolymorphic) {
-        return resourceVC != null && resourceVC.isNotFiltered() && resourceVC.isVariant() && VariantType.checkVariantType(vc, resourceVC) &&
-                (trustAllPolymorphic || !resourceVC.hasGenotypes() || resourceVC.isPolymorphicInSamples());
+    private static boolean isMatchingVariant(final VariantContext vc,
+                                             final VariantContext resourceVC,
+                                             final boolean trustAllPolymorphic,
+                                             final ResourceMatchingStrategy resourceMatchingStrategy) {
+        if (resourceVC != null && resourceVC.isNotFiltered() && resourceVC.isVariant() && VariantType.checkVariantType(vc, resourceVC) &&
+                (trustAllPolymorphic || !resourceVC.hasGenotypes() || resourceVC.isPolymorphicInSamples())) { // this is the check originally performed by VQSR
+            switch (resourceMatchingStrategy) {
+                case START_POSITION:
+                    return true;
+                case START_POSITION_AND_GIVEN_REPRESENTATION:
```

Comment:
Not sure if these strategies add any noticeable overhead if there are a lot of multiallelics, but I haven't noticed anything out of the ordinary on my runs so far.